### DISCUSSION TITLE
Supercollider File with .scd extension recognized as a Markdown file

### DISCUSSION THREAD

**Discussion by Xon77** [github-linguist/linguist#5891] [Classification] [answered]
Hello,

For a project with SuperCollider, [this file](https://github.com/Xon77/Live4Life/blob/35f3f139ec4d618e3a89896cb8cbc59c061c4222/4Live4Life_Project/_0_Init_Live4Life.scd) is recognized and colorised as as a Markdown file, while it is SuperCollider code and the extension is .scd; and I lose line numbers.

How can I get the proper colorisation and line numbers with this DuperCollider .scd file ?
Why is it recognized as a .md file ?

I had to change the extension to another with .sc to get a more adequate view with line numbers with [this other file](https://github.com/Xon77/Live4Life/blob/35f3f139ec4d618e3a89896cb8cbc59c061c4222/4Live4Life_Project/_0_Init_Live4Life.sc). But the colorisation is still wrong because I should have an extension with .scd

Many thanks

Christophe

**Comment by lildude** [comment, marked as answer]
> Why is it recognized as a .md file ?

It'll be because the `.scd` extension is shared by both languages. We don't have a heuristic to differentiate between the two so it's falling back to the classifier which is determining it looks more like our Markdown `.scd` sample than the SuperCollider sample (this one is tiny and over 10 years old).

If you've got the time and can curate a reliable regex to differentiate the two, please open a PR to add a new heuristic.

> How can I get the proper colorisation and line numbers with this DuperCollider .scd file ?

You can implement an [override](https://github.com/github/linguist/blob/master/docs/overrides.md)
